spring-rabbit/src/main/java/org/springframework/amqp/rabbit/connection/CachingConnectionFactory.java

```diff
@@ -617,7 +618,7 @@ private ChannelProxy getCachedChannelProxy(ChannelCachingConnectionProxy connect
 		else {
 			interfaces = new Class<?>[] { ChannelProxy.class };
 		}
-		return (ChannelProxy) Proxy.newProxyInstance(ChannelProxy.class.getClassLoader(),
+		return (ChannelProxy) Proxy.newProxyInstance(ClassUtils.getDefaultClassLoader(),
 				interfaces, new CachedChannelInvocationHandler(connection, targetChannel, channelList,
```

Comment:
According to Andy's comments, we don't really need it here, right?

Reply:
I'd rather have these classes loaded in the `RestartClassLoader` to be able to pick the changes up, than once sticked with the `Launcher$AppClassLoader` instead. However I might be wrong and will change this back, if you insist and have some simple explanation why we don't need this change here and there.

My understanding it is OK for `static` blocks, but if we do something at runtime would be better to rely on the context classloader. 

We can use `getApplicationContext().getClassLoader()` instead though...